internal/xray/application_res.go

```diff
@@ -0,0 +1,39 @@
+package xray
+
+import (
+	"context"
+	"fmt"
+
+	v1alpha1 "github.com/argoproj/argo-cd/pkg/apis/application/v1alpha1"
+	"github.com/derailed/k9s/internal/client"
+)
+
+// ApplicationResource represents an xray renderer.
+type ApplicationResource struct{}
+
+// Render renders an xray node.
+func (a *ApplicationResource) Render(ctx context.Context, ns string, res v1alpha1.ResourceStatus) error {
+	parent, ok := ctx.Value(KeyParent).(*TreeNode)
+	if !ok {
+		return fmt.Errorf("Expecting a TreeNode but got %T", ctx.Value(KeyParent))
+	}
+
+	gvr := gvkToGvr(res.GroupVersionKind())
+
+	root := NewTreeNode(gvr.String(), client.FQN(res.Namespace, res.Name))
+	ctx = context.WithValue(ctx, KeyParent, root)
+
+	if res.Namespace == "" {
+		parent.Add(root)
```

Comment:
A note here @derailed: these resources properly show in the tree without an additional Namespace node above them.

However, it's not possible to navigate to them or display their YAML manifest or description, as K9s fails to retrieve them. I'm guessing the current code in https://github.com/derailed/k9s/blob/master/internal/view/xray.go#L317 is not made to work with global objects.